smac/intensification/successive_halving.py

```diff
@@ -1003,12 +992,9 @@ def _launched_all_configs_for_current_stage(self, run_history: RunHistory) -> bo
         else:
             curr_insts = self.inst_seed_pairs
 
-        # _count_running_instances_for_challenger will count the running instances
-        # of the last challenger. It makes sense here, because we assume that if we
-        # moved to a new challenger, all instances have been launched for a previous
-        # challenger
-        running_instances = self._count_running_instances_for_challenger(run_history)
-        n_insts_remaining = len(curr_insts) - (self.curr_inst_idx + running_instances)
+        n_insts_remaining = len(curr_insts)
+        if self.running_challenger is not None:
+            n_insts_remaining = len(curr_insts) - self.curr_inst_idx[self.running_challenger]
```

Comment:
shouldn't this check be moved in one of the for-loops below? Or can this maybe be simplified somehow? It appears to be working, but I find this function to be very complicated now.

Reply:
Looking at this function again, doesn't it do the same thing as the function above except that the return value is either `True` or `False` and not an integer?